Subject: Use ~/.wolf4sdl for config files and saved games.
Author: Fabian Greffrath <fabian+debian@greffrath.com>

--- wolf4sdl-1.6.orig/wl_def.h
+++ wolf4sdl-1.6/wl_def.h
@@ -915,7 +915,7 @@ extern  unsigned screenofs;
 
 extern  boolean  startgame;
 extern  char     str[80];
-extern  char     configname[13];
+extern  char     configname[256];
 
 //
 // Command line parameter variables
--- wolf4sdl-1.6.orig/wl_main.cpp
+++ wolf4sdl-1.6/wl_main.cpp
@@ -10,6 +10,7 @@
 #pragma hdrstop
 #include "wl_atmos.h"
 #include <SDL_syswm.h>
+#include <errno.h>
 
 
 /*
@@ -73,7 +74,8 @@ boolean startgame;
 boolean loadedgame;
 int     mouseadjustment;
 
-char    configname[13]="config.";
+char    configdir[256];
+char    configname[256];
 
 //
 // Command line parameter variables
@@ -1871,6 +1873,17 @@ int main (int argc, char *argv[])
     CheckParameters(argc, argv);
 #endif
 
+#if defined (__unix__) && !defined(__MINGW32__)
+    /* Set up home directory */
+    int err;
+    snprintf(configdir, sizeof(configdir) - 1, "%s/.wolf4sdl/", getenv ("HOME"));
+    err = mkdir(configdir, S_IRWXU);
+    if (err == -1 && errno != EEXIST)
+        Quit ("Could not create directory \"%s\": %s\n", configdir, strerror (errno));
+#else
+    strcpy(configdir, "");
+#endif
+
     CheckForEpisodes();
 
     InitGame();
--- wolf4sdl-1.6.orig/wl_menu.cpp
+++ wolf4sdl-1.6/wl_menu.cpp
@@ -18,6 +18,7 @@
 
 extern int lastgamemusicoffset;
 extern int numEpisodesMissing;
+extern char configdir[256];
 
 //
 // PRIVATE PROTOTYPES
@@ -277,8 +278,8 @@ static int SaveGamesAvail[10];
 static int StartGame;
 static int SoundStatus = 1;
 static int pickquick;
-static char SaveGameNames[10][32];
-static char SaveName[13] = "savegam?.";
+static char SaveGameNames[10][256];
+static char SaveName[256] = "savegam?.";
 
 
 ////////////////////////////////////////////////////////////////////
@@ -1497,10 +1498,10 @@ CP_LoadGame (int quick)
 {
     FILE *file;
     int which, exit = 0;
-    char name[13];
+    char name[256];
 
 
-    strcpy (name, SaveName);
+    snprintf (name, sizeof(name) - 1, "%s%s", configdir, SaveName);
 
     //
     // QUICKLOAD?
@@ -1511,7 +1512,7 @@ CP_LoadGame (int quick)
 
         if (SaveGamesAvail[which])
         {
-            name[7] = which + '0';
+            name[strlen(configdir) + 7] = which + '0';
 #ifdef _arch_dreamcast
             DC_LoadFromVMU(name);
 #endif
@@ -1549,8 +1550,7 @@ CP_LoadGame (int quick)
         if (which >= 0 && SaveGamesAvail[which])
         {
             ShootSnd ();
-            name[7] = which + '0';
-
+            name[strlen(configdir) + 7] = which + '0';
 #ifdef _arch_dreamcast
             DC_LoadFromVMU(name);
 #endif
@@ -1676,11 +1676,11 @@ CP_SaveGame (int quick)
 {
     int which, exit = 0;
     FILE *file;
-    char name[13];
-    char input[32];
+    char name[256];
+    char input[256];
 
 
-    strcpy (name, SaveName);
+    snprintf (name, sizeof(name) - 1, "%s%s", configdir, SaveName);
 
     //
     // QUICKSAVE?
@@ -1691,7 +1691,7 @@ CP_SaveGame (int quick)
 
         if (SaveGamesAvail[which])
         {
-            name[7] = which + '0';
+            name[strlen(configdir) + 7] = which + '0';
             unlink (name);
             file = fopen (name, "wb");
 
@@ -1747,7 +1747,7 @@ CP_SaveGame (int quick)
             ShootSnd ();
 
             strcpy (input, &SaveGameNames[which][0]);
-            name[7] = which + '0';
+            name[strlen(configdir) + 7] = which + '0';
 
             fontnumber = 0;
             if (!SaveGamesAvail[which])
@@ -3142,7 +3142,7 @@ SetupControlPanel (void)
 ////////////////////////////////////////////////////////////////////
 void SetupSaveGames()
 {
-    char name[13];
+    char name[256];
 
 #ifdef _arch_dreamcast
     file_t dir;
@@ -3152,12 +3152,13 @@ void SetupSaveGames()
     dir = fs_open("/vmu/a1", O_RDONLY | O_DIR);
     x = 0;
 
-    strcpy(name, SaveName);
+    snprintf(name, sizeof(name) - 1, "%s%s", configdir, SaveName);
+
     while((dirent = fs_readdir(dir)) && x < 10)
     {
         for(int i=0; i<10; i++)
         {
-            name[7] = '0' + i;
+            name[strlen(configdir) + 7] = '0' + i;
             if(!strcmp(name, dirent->name))
             {
                 if(DC_LoadFromVMU(name) != -1)
@@ -3181,10 +3182,10 @@ void SetupSaveGames()
 
     fs_close(dir);
 #else
-    strcpy(name, SaveName);
+    snprintf(name, sizeof(name) - 1, "%s%s", configdir, SaveName);
     for(int i=0; i<10; i++)
     {
-        name[7] = '0' + i;
+        name[strlen(configdir) + 7] = '0' + i;
         const int handle = open(name, O_RDONLY | O_BINARY);
         if (handle >= 0)
         {
@@ -3616,6 +3617,7 @@ ReadAnyControl (ControlInfo * ci)
     IN_ReadControl (0, ci);
 
 
+
     if (mouseenabled && IN_IsInputGrabbed())
     {
         int mousex, mousey, buttons;
@@ -4008,7 +4010,7 @@ CheckForEpisodes (void)
     {
         strcpy (extension, "wj6");
 #endif
-        strcat (configname, extension);
+        snprintf(configname, sizeof(configname) - 1, "%sconfig.%s", configdir, extension);
         strcat (SaveName, extension);
         strcat (demoname, extension);
         EpisodeSelect[1] =
@@ -4107,7 +4109,7 @@ CheckForEpisodes (void)
     strcpy (audioext, extension);
 #endif
 
-    strcat (configname, extension);
+    snprintf (configname, sizeof(configname) - 1, "%sconfig.%s", configdir, extension);
     strcat (SaveName, extension);
     strcat (demoname, extension);
 
