diff -up Wolf4SDL-1.6-src/id_ca.cpp~ Wolf4SDL-1.6-src/id_ca.cpp
--- Wolf4SDL-1.6-src/id_ca.cpp~	2010-12-25 11:51:36.000000000 +0100
+++ Wolf4SDL-1.6-src/id_ca.cpp	2010-12-25 11:54:16.675674203 +0100
@@ -153,7 +153,7 @@ void CAL_GetGrChunkLength (int chunk)
 
 boolean CA_WriteFile (const char *filename, void *ptr, int32_t length)
 {
-    const int handle = open(filename, O_CREAT | O_WRONLY | O_BINARY);
+    const int handle = open(filename, O_CREAT | O_WRONLY | O_BINARY, 0644);
     if (handle == -1)
         return false;
 
diff -up Wolf4SDL-1.6-src/wl_main.cpp~ Wolf4SDL-1.6-src/wl_main.cpp
--- Wolf4SDL-1.6-src/wl_main.cpp~	2010-12-25 11:55:08.000000000 +0100
+++ Wolf4SDL-1.6-src/wl_main.cpp	2010-12-25 11:56:19.345674374 +0100
@@ -4,6 +4,8 @@
     #include <io.h>
 #else
     #include <unistd.h>
+    #include <sys/stat.h>
+    #include <sys/types.h>
 #endif
 
 #include "wl_def.h"
